docs/rules/no-inferrable-types.md

```diff
@@ -0,0 +1,83 @@
+# Disallows explicit type declarations for variables or parameters initialized to a number, string, or boolean. (no-inferrable-types)
+
+Explicit types where they can be easily inferred may add unnecessary verbosity.
+
+## Rule Details
+
+This rule disallows explicit type declarations on parameters, variables
+and properties where the type can be easily inferred from its value.
+
+## Options
+
+This rule has an options object:
+
+```json
+{
+    "ignoreProperties": false,
+    "ignoreParameters": false
+}
+```
+
+### Default
+
+When none of the options are truthy, the following patterns are valid:
+
+```ts
+const foo = 5;
+const bar = true;
+const baz = 'str';
+
+class Foo {
+    prop = 5;
+}
+
+function fn(a = 5, b = true) {
```

Comment:
Is `function fn(a: number, b: string) {}` valid?

Reply:
Yup, will add it to the docs and tests tomorrow.

We only check assignments so normal type annotations get through fine.